### PR TITLE
Refactored the iOS Scrolling Simulation

### PR DESCRIPTION
**Pull request by HansMuller**
Removed the SimulationGroup class and rewrote BouncingScrollSimulation without it.

This change addresses a need that came up for https://github.com/flutter/flutter/pull/9710

### CODE REVIEW

File: packages/flutter/lib/src/physics/friction_simulation.dart [hunk 1]
```diff
@@ -71,6 +71,20 @@ class FrictionSimulation extends Simulation {
   @override
   double dx(double time) => _v * math.pow(_drag, time);
 
+  /// The value of [x] at `double.INFINITY`;
+  double get finalX => _x - _v / _dragLog;
```
Comment (abarth) [inline packages/flutter/lib/src/physics/friction_simulation.dart]: There's a "terminal velocity" joke in here somewhere

Comment (HansMuller) [inline packages/flutter/lib/src/physics/friction_simulation.dart]: :-)

Comment (Hixie) [inline packages/flutter/lib/src/physics/friction_simulation.dart]: Trailing semicolon should presumably be a period.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Gold star

**Pull request merged**